### DISCUSSION TITLE
please implement multilingual MultiplexedOCR

### DISCUSSION THREAD

**Discussion by mobassir94** [PaddlePaddle/PaddleOCR#6494] [Ideas & Features]
hi,
you are claiming this is a multilingual ocr project but in general i don't think it really is yet. your pipeline doesn't have multi head recognition branch for multiple languages. i think you can train one end to end model like this : https://github.com/facebookresearch/MultiplexedOCR with different head working for different languages all together. this is a classic and IMHO the best multilingual ocr possible,,but the authors of that work is not responsive and it's hard to work on such project with little or no documentation,,maybe if your team can train similar multilingual ocr and include that in paddleocr then imho it will be great future for paddleocr's multilingual ocr journey

**Comment by Evezerest** [comment]
hi Mobassir, thanks for your attention to PaddleOCR.
We do not provide a model that recognizes more than two languages ​​at the same time because we are not quite sure if there is such a need.
I'm very curious in what circumstances do you need such a model, and which languages ​​are needed.
Currently, you can use multiple models simultaneously to recognize multiple languages.
LFTYR

**Reply by mobassir94** [reply]
there are many samples like this possible : https://image.shutterstock.com/image-vector/young-people-chatting-foreign-languages-600w-1723737016.jpg 
paddleocr will fail on such samples,,currently no public ocr is actually real multilingual,which is sad,i doubt these ocr's will become kind of unusable after few years,there are multilingual documents possible,most public ocr including paddle will fail on such situation,
for example, i have prepared 2 datasets : 
1. https://www.kaggle.com/datasets/mobassir/tafsir-ibn-kathir-bangla
2.  https://www.kaggle.com/datasets/mobassir/tafsir-ibn-kathir-in-english

most pages/images there will contain arabic+english mixed or arabic+bangla mixed,,this is just an example,,there are many more possible domain exist that contains code mixed samples.

**Currently, you can use multiple models simultaneously to recognize multiple languages.** 
this is what,everyone else has been doing,this is not a good solution, read the MultiplexedOCR paper,authors of that paper described there why this approach is not a good solution for multilingual ocr and why we need e2e solution with multiple heads(each head for each language)


another disadvantage of multiple model based solution is you need more models(more computation,more gpu vram) and also need one extra classifier for classifying the language of cropped/detected word image (this is complex and brute force like solution) which is not good IMHO
